### ISSUE TITLE
house_number when parsing

### ISSUE THREAD

**Issue by sberryman** [gosom/address-parser-go-rest#1]
First, thanks for creating this project and sharing it!

I'm working on parsing some addresses from the Bureau of the Fiscal Service:
https://www.fiscal.treasury.gov/surety-bonds/list-certified-companies.html
Excel file: https://www.fiscal.treasury.gov/files/surety-bonds/list-certified-companies.xlsx

I've noticed that the address for quite a few of these businesses have the postal code in 5+ format with a space. This causes the 4 digits for the extended postal code to be parsed as `house_number`.

Example:
```shell
curl -X 'POST' \
  'http://localhost:8080/parse' \
  -H 'accept: application/json' \
  -H 'Content-Type: application/json' \
  -d '{
  "address": "2800 South Taylor Drive, P.O. Box 58, Sheboygan, WI 53082 - 0058",
   "title_case": true
}'
```

Result:
```json
{
    "house_number": "0058",
    "road": "South Taylor Drive",
    "po_box": "P.o. Box 58",
    "postcode": "53082",
    "city": "Sheboygan",
    "state": "Wi"
}
```

Libpostal outputs two components for the above address for `house_number` which is annoying but would at least allow me to grab the first item.

I'm not sure the best way to handle this, I'm also not sure if every component could have multiple outputs from the parser?

Thoughts?

**Comment by gosom** [comment]
hello,

you are right libpostal outputs two components for `house_number`  in this example.

In your example:

```
{house_number 2800} {road south taylor drive} {po_box p.o. box 58} {city sheboygan} {state wi} {postcode 53082} {house_number 0058}
```

probably I should add a field `house_number_2`  ?
```
curl -X 'POST'   'http://localhost:8080/parse'   -H 'accept: application/json'   -H 'Content-Type: application/json'   -d '{
  "address": "2800 South Taylor Drive, P.O. Box 58, Sheboygan, WI 53082 - 0058",
   "title_case": true
}'
{"house_number":"2800","house_number2":"0058","road":"South Taylor Drive","po_box":"P.o. Box 58","postcode":"53082","city":"Sheboygan","state":"Wi"}
```

alternatively we could concatenate these space separated ? 

```
 curl -X 'POST'   'http://localhost:8080/parse'   -H 'accept: application/json'   -H 'Content-Type: application/json'   -d '{
  "address": "2800 South Taylor Drive, P.O. Box 58, Sheboygan, WI 53082 - 0058",
   "title_case": true
}'

{"house_number":"2800 0058","road":"South Taylor Drive","po_box":"P.o. Box 58","postcode":"53082","city":"Sheboygan","state":"Wi"}
```

both solutions are ugly. 

any other idea?

**Comment by sberryman** [comment]
Yeah, no really good solutions. Maybe the struct is `[]string`? I don't think that is any better though. 

#2 is probably best for me but that is only speaking from experience with USA addresses. Maybe there is another `ambiguous` field which is just an array of:

```golang
type ambiguous struct {
    component string
    val string
}
```

Just thinking out loud here... That way the data isn't "lost" and the dev can still look through the ambiguous array? Basically any component that appears more than once has the duplicate keys added to ambiguous?

**Comment by gosom** [comment]
a. For now I added only the first occurence of the house_number . For your case it will work (and in majority of cases I believe, since this is an edge case.

b. based on your idea with ambiguous . Why not adding a field 
```
libpostal_components: [ {...}] 
```
which contains the same data as libpostal ? 

This way nothing is "lost"

**Comment by gosom** [comment]
I will close this issue for now and will open a new one to add the libpostal_components: [{...}] . 

thanks for the bug report . 

The issue with the addresses you posted should have been resolved in v1.0.1

```
docker run -p 8080:8080 gosom/address-parser-go-rest:v1.0.1
```

Result
```
curl -X 'POST'   'http://localhost:8080/parse'   -H 'accept: application/json'   -H 'Content-Type: application/json'   -d '{
  "address": "2800 South Taylor Drive, P.O. Box 58, Sheboygan, WI 53082 - 0058",
   "title_case": true
}'
{"house_number":"2800","road":"South Taylor Drive","po_box":"P.o. Box 58","postcode":"53082","city":"Sheboygan","state":"Wi"}
```

**Comment by sberryman** [comment]
Excellent! Thanks for the really quick response
